parser/parser_test.go

```diff
@@ -72,12 +73,72 @@ func TestParser(t *testing.T) {
 		{"", "USE system", true, false, &UseStatement{
 			Keyspace: "system",
 		}},
+
+		// Reads from tables named similarly to system tables (not handled)
 		{"", "SELECT count(*) FROM local", false, true, nil},
 		{"", "SELECT count(*) FROM peers", false, true, nil},
 		{"", "SELECT count(*) FROM peers_v2", false, true, nil},
-		{"", "INSERT INTO system.local (key, rpc_address) VALUES ('local1', '127.0.0.1')", false, false, nil},
-		{"", "UPDATE system.local SET rpc_address = '127.0.0.1' WHERE key = 'local'", false, false, nil},
-		{"", "DELETE rpc_address FROM system.local WHERE key = 'local'", false, false, nil},
+
+		// Mutations to system tables (not handled)
+		{"", "INSERT INTO system.local (key, rpc_address) VALUES ('local1', '127.0.0.1')", false, true, nil},
+		{"", "UPDATE system.local SET rpc_address = '127.0.0.1' WHERE key = 'local'", false, true, nil},
+		{"", "DELETE rpc_address FROM system.local WHERE key = 'local'", false, true, nil},
+
+		// Mutations that use the functions whose values change e.g. uuid(), now() (not idempotent)
+		{"", "INSERT INTO ks.table (key, value) VALUES ('k1', uuid())", false, false, nil},
+		{"", "INSERT INTO ks.table (key, value) VALUES ('k1', now())", false, false, nil},
+		{"", "INSERT INTO ks.table (key, value) VALUES ('k1', nested(now()))", false, false, nil},
+		{"", "INSERT INTO ks.table (key, value) VALUES ('k1', nested(uuid()))", false, false, nil},
+		{"", "UPDATE ks.table SET v = uuid() WHERE k = 1", false, false, nil},
+		{"", "UPDATE ks.table SET v = now() WHERE k = 1", false, false, nil},
```

Comment:
What about DELETE with `now()`? Should we add that just to be safe?

Reply:
Yes, this is correct. Use of `now()` or `uuid()` in a "where" clause is not idempotent. Good catch.